jerry-core/ecma/builtin-objects/ecma-builtin-regexp-prototype.c

```diff
@@ -63,156 +63,157 @@ ecma_builtin_regexp_prototype_compile (ecma_value_t this_arg, /**< this argument
                                        ecma_value_t pattern_arg, /**< pattern or RegExp object */
                                        ecma_value_t flags_arg) /**< flags */
 {
-  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
-
   if (!ecma_is_value_object (this_arg)
       || !ecma_object_class_is (ecma_get_object_from_value (this_arg), LIT_MAGIC_STRING_REGEXP_UL))
   {
-    ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("Incomplete RegExp type"));
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Incomplete RegExp type"));
   }
-  else
-  {
-    uint16_t flags = 0;
 
-    if (ecma_is_value_object (pattern_arg)
-        && ecma_object_class_is (ecma_get_object_from_value (pattern_arg), LIT_MAGIC_STRING_REGEXP_UL))
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
```

Comment:
This declaration is way too far away from the first use. (And because the first access to this variable is actually an assignment, this initialisation is superfluous.)